instrumentation/java-concurrent/src/main/java/io/opentelemetry/instrumentation/auto/javaconcurrent/KotlinProbeInstrumentation.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.instrumentation.auto.javaconcurrent;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+import com.google.auto.service.AutoService;
+import io.grpc.Context;
+import io.opentelemetry.javaagent.tooling.Instrumenter;
+import java.util.HashMap;
+import java.util.Map;
+import kotlin.coroutines.Continuation;
+import kotlin.coroutines.CoroutineContext;
+import kotlin.jvm.functions.Function2;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@AutoService(Instrumenter.class)
+public class KotlinProbeInstrumentation extends Instrumenter.Default {
```

Comment:
I think it'd make sense for us to rename `kotlin-testing` to `kotlin-coroutines` and add this instrumentation there too since this isn't instrumenting the JDK itself.

Reply:
Are you proposing to move this instrumentation and test to a new <root>/instrumentation/kotlin-coroutines module?  That makes sense to me.  Please confirm my understanding and I'll work on that refactoring.

Reply:
Yup that's what I'm thinking

Reply:
OK, I'll work on that today.  Thanks!